default/AI/ColonisationAI.py

```diff
@@ -103,6 +105,16 @@ def rate_planetary_piloting(pid):
         return 0.0
     return rate_piloting_tag(this_spec.tags)
 
+def calc_supply_tech_range():
+    supply_range = 0
+    for tech in AIDependencies.supply_range_techs:
+        if tech_is_complete(tech):
+            supply_distance += AIDependencies.supply_range_techs[tech]
+
+
+def get_supply_tech_range():
```

Comment:
You can use decorator for such case to avoid creating new variable.
```python
from functools import wraps


def turn_cache(function):
    """
    Cache value for current turn. Wraps only functions with out arguments.
    """
    data = [None, None]  # last turn, cached value
    @wraps
    def wrapper():
        if data[0] != fo.currentTurn():
            data[0] = fo.currentTurn()
            data[1] = function()
        return data[1]
    return wrapper


@turn_cache
def get_supply_tech_range():
```